From 18ca88e81c30c262ed7ed555c91cb1283b78d4a5 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Tue, 20 Apr 2021 06:57:59 +0300
Subject: [PATCH 254/302] Revert "LF-1409-05: media: isi: config ISI HW driver
 as a separate driver"

This reverts commit f8ba82053cbe6ba09953d7106cd5c178bddd30c3.
---
 drivers/staging/media/imx/Kconfig  | 9 ---------
 drivers/staging/media/imx/Makefile | 3 +--
 2 files changed, 1 insertion(+), 11 deletions(-)

diff --git a/drivers/staging/media/imx/Kconfig b/drivers/staging/media/imx/Kconfig
index d12dc6a9ec53..674a4723fad1 100644
--- a/drivers/staging/media/imx/Kconfig
+++ b/drivers/staging/media/imx/Kconfig
@@ -56,15 +56,6 @@ config IMX8_MEDIA_DEVICE
 	  This media device is a virtual device which used to manage
 	  all modules in image subsystem of imx8qxp/qm platform.
 
-config IMX8_ISI_HW
-	tristate "IMX8 Image Sensor Interface hardware driver"
-	depends on VIDEO_V4L2 && VIDEO_V4L2_SUBDEV_API
-	default y
-	help
-	  ISI hardware driver is used to export functions to config
-	  ISI registers and it is shared by isi capture and mem2mem
-	  driver
-
 config IMX8_ISI_CORE
 	tristate "IMX8 Image Sensor Interface Core Driver"
 	depends on IMX8_ISI_CAPTURE && IMX8_ISI_M2M
diff --git a/drivers/staging/media/imx/Makefile b/drivers/staging/media/imx/Makefile
index 70866ac05924..036e302d2ec1 100644
--- a/drivers/staging/media/imx/Makefile
+++ b/drivers/staging/media/imx/Makefile
@@ -6,7 +6,7 @@ imx6-media-objs := imx-media-dev.o imx-media-internal-sd.o \
 	imx-ic-common.o imx-ic-prp.o imx-ic-prpencvf.o imx-media-vdic.o \
 	imx-media-csc-scaler.o
 
-imx8-capture-objs := imx8-isi-core.o
+imx8-capture-objs := imx8-isi-core.o imx8-isi-hw.o
 
 imx6-media-csi-objs := imx-media-csi.o imx-media-fim.o
 
@@ -22,7 +22,6 @@ obj-$(CONFIG_IMX8_MEDIA_DEVICE)   += imx8-media-dev.o
 obj-$(CONFIG_IMX8_ISI_CORE) += imx8-capture.o
 obj-$(CONFIG_IMX8_ISI_CAPTURE) 	  += imx8-isi-cap.o
 obj-$(CONFIG_IMX8_ISI_M2M)	+= imx8-isi-m2m.o
-obj-$(CONFIG_IMX8_ISI_HW) 	  += imx8-isi-hw.o
 obj-$(CONFIG_IMX8_MIPI_CSI2) 	  += imx8-mipi-csi2.o
 obj-$(CONFIG_IMX8_MIPI_CSI2_SAM)  += imx8-mipi-csi2-sam.o
 obj-$(CONFIG_IMX8_PARALLEL_CSI)	  += imx8-parallel-csi.o
-- 
Regards,

Laurent Pinchart

